RATEST-298: Fix E2E Test: Appointment management workflow

Irenyak1 opened the pull request:
Ticket ID -> [RATEST-298](https://issues.openmrs.org/browse/RATEST-298)

Description -> E2E Test: Appointment management workflow

Legacy tests changed to BDD framwork

    AddPatientAppointmentTest
    BookRequestAppointmentTest
    DeleteRequestAppointmentTest
    AddEditAppointmentBlockTest
    DeleteAppointmentBlockTest
    ManageProviderSchedulesTest

Irenyak1 commented:
Hello @sherrif10 , @kdaud  please make some time and review my PR here. I am looking into the build failure.

sherrif10 commented:
You could start from the build failures

Irenyak1 commented:
> You could start from the build failures

That is what I am doing.  The issue is coming from the instability of the xpath selector her 
https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a22cb2fcca3b39e542028af477bf2b8c15b7616/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java#L20

Irenyak1 commented:
@sherrif10 this is serious! Let me look into the failures.

sherrif10 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointments.feature:
```diff
@@ -1 +1,70 @@
+		When a user clicks Appointment scheduling app from the home page
+		Then the system loads the appointment scheduling page
+		And a user clicks on Manage Provider Schedules app
+		And the system loads Manage Provider Schedules page
```
These workflow are too many  which looks a bit not clear. Try to make sure to minimize the user stories as much as possible to make them meanfull. Suppose this workflow land on you after 10 years ago what would you think. Make sure you refactor them to reduce on the user stories but keeping them appropriate, readable and making meaning.

Irenyak1 replied:
@sherrif10 actually those work flows build on top of each other. One must have created an appointment request as the initial step. that is why I have put that as the background. Then from there, the user has to create an appointment block on which the user builds to create an appointment. and then the edit and delete also definitely have to build on top of a created appointment block. So that is why the four workflows are bound under one scenario because they have to rely on a created appointment block.

Irenyak1 replied:
If I am to separate them, then it means that in every scenario I have to create an appointment block. Would that be a better implementation of this @sherrif10 . Let me know so I can change to that.

sherrif10 commented:
@Irenyak1  According to your server errros, you are messing up with some tests classes. Can you cross check appointmentTest classes whether they are worked upon well.

Irenyak1 commented:
> @Irenyak1 According to your server errros, you are messing up with some tests classes. Can you cross check appointmentTest classes whether they are worked upon well.

I have fixed it. It happened when I updated my code with upstream changes.

mherman22 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature:
```diff
@@ -0,0 +1,70 @@
+		And a user fills in the request details
+		And a user clicks save button
+		Then the system adds appointment request into the appointment requests table
+		
```
So many steps here need to be combined @Irenyak1. 
Refer to

mherman22 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature:
```diff
@@ -0,0 +1,70 @@
+		And a user searches for a patient
+		And a user clicks on the first patient
+		And the system loads Manage Appointments page
+		And a user enters service
```
Instead of having 
```
And a user enters service
And a user searches appointment
```
Cant we have one step as `And a user searches appointment`

mherman22 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature:
```diff
@@ -0,0 +1,70 @@
+		And the system loads Appointment Blocks page
+		And a user clicks on an appointment block
+		And a user clicks on Delete
+		And a user confirms Delete Appointment Block
```
Same here, cant we have one step as  `And a user deletes appointment block`
and finish with `Then the systems deletes appointment block from the appointment Blocks table`

mherman22 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature:
```diff
@@ -0,0 +1,70 @@
+		And a user searches for a patient
+		And a user clicks on the first patient
+		And the system loads Manage Appointments page
+		And a user enters service
```
We can combine these aswell 
```
And a user enters service
And a user searches appointment
```
Into `And a user searches appointment`

sherrif10 replied:
No objection at all.

sherrif10 commented:
Try as much as possible to combine the steps. There is a tendency of
duplicate which we might not like to fall into.

On Fri, Jan 21, 2022, 10:40 AM Herman Muhereza ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In
> qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature
> <https://github.com/openmrs/openmrs-contrib-qaframework/pull/323#discussion_r789417190>
> :
>
> > +		And the system loads Appointment Blocks page
> +		And a user clicks on an appointment block
> +		And a user clicks on Delete
> +		And a user confirms Delete Appointment Block
> +		Then the systems deletes appointment block from the appointment Blocks table
> +		
> +	@selenium
> +	@appointment
> +	Scenario: Delete Appointment Request
> +		When a user clicks Appointment scheduling app from the home page
> +		Then the system loads the appointment scheduling page
> +		And a user clicks on Manage Appointments app
> +		And a user searches for a patient
> +		And a user clicks on the first patient
> +		And the system loads Manage Appointments page
> +		And a user enters service
>
> Instead of having
>
> And a user enters service
> And a user searches appointment
>
> Cant we have one step as And a user searches appointment
> ------------------------------
>
> In
> qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature
> <https://github.com/openmrs/openmrs-contrib-qaframework/pull/323#discussion_r789419772>
> :
>
> > +		And a user searches appointment
> +		And a user clicks on an appointment request
> +		And a user clicks on book appointment icon
> +		And a user clicks the Save button
> +		And the system saves the scheduled appointment
> +		And a user enters patient name
> +		And a user clicks on the first patient
> +		Then the system loads the Manage Appointments page
> +		# User story: Delete Appointment Block
> +		When a user clicks Appointment scheduling app from the home page
> +		Then the system loads the appointment scheduling page
> +		And a user clicks on the Manage Provider Schedules app
> +		And the system loads Appointment Blocks page
> +		And a user clicks on an appointment block
> +		And a user clicks on Delete
> +		And a user confirms Delete Appointment Block
>
> Same here, cant we have one step as And a user deletes appointment block
> and finish with Then the systems deletes appointment block from the
> appointment Blocks table
> ------------------------------
>
> In
> qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature
> <https://github.com/openmrs/openmrs-contrib-qaframework/pull/323#discussion_r789421004>
> :
>
> > +		And a user clicks on the Manage Provider Schedules app
> +		And the system loads Appointment Blocks page
> +		And a user clicks on an appointment block
> +		And a user clicks on Edit
> +		And the system loads Edit Appointment Block section
> +		And a user edits the Appointment block
> +		And a user clicks on Save button
> +		Then the systems saves the edited appointment block
> +		# User story: Add Appointment
> +		When a user clicks Appointment scheduling app from the home page
> +		Then the system loads the appointment scheduling page
> +		And a user clicks on Manage Appointments app
> +		And a user searches for a patient
> +		And a user clicks on the first patient
> +		And the system loads Manage Appointments page
> +		And a user enters service
>
> We can combine these aswell
>
> And a user enters service
> And a user searches appointment
>
> Into And a user searches appointment
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-contrib-qaframework/pull/323#pullrequestreview-859189258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALOKS6DTCLCEII2MKBJV7A3UXEEVPANCNFSM5MJWJAVA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Irenyak1 commented:
@sherrif10 , @kdaud this test is so unstable, It runs fine on my local machine and right here it fails. I have tried to fix areas being pointed out but thereafter it jumps to another. Could there be someone to help me so we can sync and look into it?

sherrif10 commented:
Looking through the error logs https://github.com/openmrs/openmrs-contrib-qaframework/runs/4918313641?check_suite_focus=true   . This feature have a problem at ✽.a user clicks on the Schedule Appointment link from the Patient.

can you cross check with your feature file number 60

Irenyak1 commented:
> Looking through the error logs https://github.com/openmrs/openmrs-contrib-qaframework/runs/4918313641?check_suite_focus=true . This feature have a problem at ✽.a user clicks on the Schedule Appointment link from the Patient.
> 
> can you cross check with your feature file number 60

it passes so well on my local machine.  I am looking into it again.

Muta-Jonathan commented:
@Irenyak1 ....```
Booking An Appointment(Appointment Management): Error communicating with the remote browser. It may have died.(..)
  Booking An Appointment(Appointment Management): Error communicating with the remote browser. It may have died.(..)
```

test in error.....
try using the **wait selenium ....method**

Irenyak1 commented:
> @Irenyak1 ....
> 
> ```
> Booking An Appointment(Appointment Management): Error communicating with the remote browser. It may have died.(..)
>   Booking An Appointment(Appointment Management): Error communicating with the remote browser. It may have died.(..)
> ```
> 
> test in error..... try using the **wait selenium ....method**

@Muta-Jonathan I am trying to look into this. Thanks.

kdaud commented:
@Irenyak1 have you taken a look at the build failure? https://github.com/openmrs/openmrs-contrib-qaframework/runs/5033813851?check_suite_focus=true

Irenyak1 commented:
@sherrif10 and @kdaud could someone help me rerun this build. I have realised that the Appointment workflow keeps failing at different sections. I don't know what really is causing this. I have tried rerunning it locally and it is passing well. Just don't know what is causing the instability in the CI build.

Irenyak1 commented:
@kdaud  could you trigger a rerun for this build.  I see the cause of failure was an obstruction of an element.

kdaud commented:
> @kdaud could you trigger a rerun for this build.

have done the needful !

kdaud commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -15,14 +15,14 @@
-	private static final By SAVE = By.xpath("//div[@id='confirmAppointment']/div[2]/button[2]");
+	private static final By CANCEL_BUTTON = By.cssSelector("#searchButtons button");
+	private static final By SEARCH = By.cssSelector("#searchButtons button.confirm");
+	private static final By APPOINTMENT = By.cssSelector("#appointmentTable div:nth-child(2) div:nth-child(1) div:nth-child(1) div:nth-child(1) div:nth-child(2) div:nth-child(2) div:nth-child(1)");
```
we can get better element locator that can hardly break!

Irenyak1 replied:
These elements do not have other unique ways to access them unfortunately.

kdaud commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -15,14 +15,14 @@
+	private static final By SEARCH = By.cssSelector("#searchButtons button.confirm");
+	private static final By APPOINTMENT = By.cssSelector("#appointmentTable div:nth-child(2) div:nth-child(1) div:nth-child(1) div:nth-child(1) div:nth-child(2) div:nth-child(2) div:nth-child(1)");
+	private static final By NEXT = By.cssSelector("#selectAppointment button.confirm");
+	private static final By SAVE = By.cssSelector("#confirmAppointment div:nth-child(3) button.confirm.right");
```
even here

Irenyak1 replied:
@kdaud  Let me look out for other selectors

sherrif10 commented:
Resolve all merge conflicts.

kdaud commented:
@Irenyak1 worth resolving the conflicting files!

Irenyak1 commented:
> @Irenyak1 worth resolving the conflicting files!

Yes @kdaud  I have fixed that. I haven't just pushed a commit, something I am still fixing. Thanks

Irenyak1 commented:
@sherrif10  could you please help me rerun this build? The error logs points to an element that I think is not really a problem.

sherrif10 commented:
@Irenyak1  Running all the builds or specific workflow ?

Irenyak1 commented:
> @Irenyak1 Running all the builds or specific workflow ?

@sherrif10 all the build.

sherrif10 commented:
You can always trigger empty commit to re run your builds.

sherrif10 commented:
@Irenyak1  i hope you looking into the build failures currently.

Irenyak1 commented:
> @Irenyak1 i hope you looking into the build failures currently.

This is the pointer in the failure logs. `Tests in error: 
  verifyNurseModulesAvailableOnHomePage(org.openmrs.contrib.qaframework.legacy.LoginTest): Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.`

Irenyak1 commented:
@sherrif10 are there more reviews on this?

kdaud commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -15,14 +15,14 @@
-	private static final By NEXT = By.cssSelector("#selectAppointment > button.confirm");
 	private static final By SAVE = By.xpath("//div[@id='confirmAppointment']/div[2]/button[2]");
+	private static final By CANCEL_BUTTON = By.cssSelector("#searchButtons button");
+	private static final By SEARCH = By.cssSelector("#searchButtons button.confirm");
```
is `SEARCH` a button or a link?

kdaud commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -15,14 +15,14 @@
 	private static final By SAVE = By.xpath("//div[@id='confirmAppointment']/div[2]/button[2]");
+	private static final By CANCEL_BUTTON = By.cssSelector("#searchButtons button");
+	private static final By SEARCH = By.cssSelector("#searchButtons button.confirm");
+	private static final By NEXT = By.cssSelector("#selectAppointment button.confirm");
```
is `NEXT` a button or a link?

Irenyak1 replied:
> is `NEXT` a button or a link?

@kdaud it is a button

kdaud replied:
how can one know it's a button?

kdaud replied:
Giving it a distinctive name like `NEXT_BUTTON` would be better!

sherrif10 replied:
How about `CONFIRM_BUTTON`

Irenyak1 replied:
> Why are we combining all these in one class, contrary to the original design which had separated them into different classes?

@dkayiwa I have not understood this question. Do you mean combing the different test into one? If yes, we wanted to create an E2E workflow, hence combining the different test classes.

kdaud commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -14,38 +14,38 @@
-	private static final By BOOK_APPOINTMENT = By.cssSelector("i.icon-calendar:nth-child(1)");
-	private static final By YES_DELETE_REQUEST = By.xpath("//div[@id='confirm-cancel-appointment-request']/div[2]/button");
+	private static final By SEARCH_BUTTON = By.cssSelector("#searchButtons button.confirm");
+	private static final By NEXT_BUTTON = By.cssSelector("#selectAppointment button.confirm");
```
I think we can have this renamed to `CONFIRM_BUTTON` instead of `NEXT_BUTTON`

Irenyak1 replied:
@sherrif10 , @kdaud  no it is not the final button that confirms the saving of an appointment, it is a NEXT button that when clicked leads to another stage where finally the save button is clicked to save the appointment. I still suggest that we maintain the NEXT_BUTTON as opposed to CONFIRM_BUTTON.

sherrif10 replied:
Do you mind pointing us to the last confirm button url

kdaud reviewed (approved):
cc: @sherrif10

Irenyak1 commented:
@sherrif10, do you still have reviews for this or it can be merged into master?

sherrif10 commented:
Sorry @Irenyak1 . Let me first re run these workflows first.

sherrif10 commented on qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageAppointmentsPage.java:
```diff
@@ -14,38 +14,38 @@
+	private static final By SEARCH_BUTTON = By.cssSelector("#searchButtons button.confirm");
+	private static final By NEXT_BUTTON = By.cssSelector("#selectAppointment button.confirm");
+	private static final By BOOK_APPOINTMENT_ICON = By.cssSelector("i.icon-calendar:nth-child(1)");
+	private static final By YES_DELETE_REQUEST_BUTTON = By.xpath("//div[@id='confirm-cancel-appointment-request']/div[2]/button");
```
What do you think of `CONFIRM_CANCEL_APPOINTMENT` instead of `YES_DELETE_REQUEST_BUTTON `

Irenyak1 replied:
> What do you think of `CONFIRM_CANCEL_APPOINTMENT` instead of `YES_DELETE_REQUEST_BUTTON `

Could I add button on `CONFIRM_CANCEL_APPOINTMENT_BUTTON` instead to let someone know it is a button? I am following the reviews made earlier about variable names.

sherrif10 replied:
I  think even though you use  CONFIRM_CANCEL_APPOINTMENT_BUTTON  . it still remains a button. does it change.

Irenyak1 replied:
> I think even though you use `CONFIRM_CANCEL_APPOINTMENT_BUTTON `. it still remains a button. does it change.oK Why are you calling it `YES_DELETE_REQUEST_BUTTON`

@sherrif10 you had suggested that I name it `CONFIRM_CANCEL_APPOINTMENT ` without the word button so I suggested that I add button to make it `CONFIRM_CANCEL_APPOINTMENT_BUTTON ` for clarity. I am not objecting. I called it `YES_DELETE_REQUEST_BUTTON` because the button text is `YES`

sherrif10 replied:
`CONFIRM_CANCEL_APPOINTMENT_BUTTON  `  looks much better.

Irenyak1 replied:
> `CONFIRM_CANCEL_APPOINTMENT_BUTTON ` looks much better

@sherrif10 Ok then let me fix this and push a commit.

sherrif10 reviewed (approved):
Thanks, @Irenyak1. This looks morethan enough to merge. cc @dkayiwa .

kdaud commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+	}
+
+	@And("a user fills in the request details")
+	public void captureRequestDetails() {
```
the method name need to be revised to reflect the request being captured?

Irenyak1 commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+	}
+
+	@And("a user clicks on Delete appointment block")
+	public void clickOnDelete() throws InterruptedException {
```
@kdaud I have attended to both reviews and made a commit.

kdaud reviewed (approved):
cc: @dkayiwa

dkayiwa commented:
Why are we combining all these in one class, contrary to the original design which had separated them into different classes?

Irenyak1 commented:
@dkayiwa we tried combining related tests into one E2E workflow, hence combining the different test classes.

dkayiwa commented:
Is this better than the original design which refactored the tests in different classes?

Irenyak1 commented:
> Is this better than the original design which refactored the tests in different classes?

@dkayiwa  I think it is  better since one test class is able to contain all tests involved in a workflow. It is also something I remember was discussed in the squad calls where there was a preference of testing according to workflows. For example incase a patient wanted to make an appointment. The test class would combine all steps involved in making an appointment.

dkayiwa commented:
How will the error message look like if for instance `Delete Appointment Block ` fails but all the others pass?

kdaud commented:
The QA team agreed on transforming legacy tests into bdd tests so that we have a qaframework module built on the same framework architecture and that we combine legacy tests that rotates on a similar feature of refApp into a workflow. 

> How will the error message look like if for instance Delete Appointment Block fails but all the others pass?

This is tracked via cucumber report that tells out which gherkin step has passed/failed on every build plan.

dkayiwa commented:
Did you look into why all Firefox Tests are failing?

dkayiwa commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+	public void captureAppointmentBlockDetails() {
+		manageProviderSchedulesPage.selectLocation("Laboratory");
+		manageProviderSchedulesPage.clickOnNextWeekday();
+		manageProviderSchedulesPage.selectLocationBlock("Laboratory");
```
Some text values are hard coded while others are in well defined constants. Was this intentional?

Irenyak1 replied:
@dkayiwa sure I did that but now I see it was not proper. Let me align defining constants.

dkayiwa commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
```
How does the above prove that the appointment request has been added to the table?

Irenyak1 replied:
@dkayiwa  this proves that the appointment request list is not empty after adding the request since the test assumes that it is the only request that has been saved.

dkayiwa commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+
+	@And("the system loads Manage Provider Schedules page")
+	public void systemLoadsManageProviderSchedulesPage() {
+		assertTrue(textExists("Manage Appointment Blocks"));
```
Is this the manager provider schedules or manage appointment blocks page?

Irenyak1 replied:
@dkayiwa it is  manage provider schedules  page. However, this page together with  appointment blocks page are both loaded at the same url "/appointmentschedulingui/scheduleProviders.page"  and both contain "Manage Appointment Blocks" on display. It is only they implement different functionalities.

dkayiwa commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+
+	@Then("the system saves the appointment block")
+	public void systemSavesAppointmentBlockDetails() {
+		assertTrue(manageProviderSchedulesPage.containsText(FIRST_SERVICE_NAME));
```
How does this prove that the appointment block has been saved?

Irenyak1 replied:
@dkayiwa  before saving the appointment block, Manage Provider Schedules Page does not contain the service name until a block is saved, then it displays the service that has been blocked for appointment.

dkayiwa commented on qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java:
```diff
@@ -0,0 +1,240 @@
+
+	@And("a user clicks on the Manage Provider Schedules app")
+	public void loadAppointmentBlocksPage() {
+		appointmentBlocksPage = (AppointmentBlocksPage ) appointmentSchedulingPage.goToAppointmentBlock();
```
Is this manage provider schedules or appoint blocks page?

Irenyak1 replied:
@dkayiwa  I hope t[his comment ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/323#discussion_r822333268) will suffice.

Irenyak1 commented:
> Did you look into why all Firefox Tests are failing?

@dkayiwa  all firefox tests are now passing.

Irenyak1 commented:
@sherrif10 , @kdaud  . @dkayiwa  do we still have reviews on this issue?

sherrif10 commented:
Have you attended to Mr @dkayiwa  comments.

Irenyak1 commented:
> Have you attended to Mr @dkayiwa comments.

@sherrif10 I attended to all of them.

sherrif10 commented on qaframework-bdd-tests/src/test/resources/features/refapp-2.x/13-appointment/appointment.feature:
```diff
@@ -0,0 +1,66 @@
+    Then the system loads the appointment scheduling page
+    And a user clicks on Manage Provider Schedules app
+    And the system loads Manage Provider Schedules page
+    And a user fills in the details of the appointment block
```
@Irenyak1  This line is breaking the ci with these logs https://ci.openmrs.org/browse/CONTRIB-QA-FRAMEWORK-1403/log  . Cross check with the method named `[clickOnNextWeekDay](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageProviderSchedulesPage.java#L20)`  and see the elements used whether they meet the standards. Then update it with your feature files.

Irenyak1 replied:
@sherrif10 Let me look into it.

kdaud commented:
@Irenyak1 any updates?

kdaud commented:
@Irenyak1 how far with fixing this test?

Irenyak1 commented:
> @Irenyak1 how far with fixing this test?

@kdaud  I am still looking into it. unfortunately it runs well locally but when it got merged it broke the build. The part it is pointing to looks fine, though I am finding a way of changing selectors or stopping the overshadowing element.

Irenyak1 commented:
Hello @kdaud , @sherrif10 , @Muta-Jonathan , @dkayiwa , @mherman22 , I have been investigating this test and I have found the probable cause of its failure.  
When the test is done and the test patient is deleted(hoping he/she is deleted) I would expect everything about the test patient to be deleted, however, if an appointment was scheduled, it remains in the system, it does not get deleted. Because of this. when the test is rerun, at the point of clicking a day, in case the day contains a scheduled appointment, the day click is not successful since the click hits the scheduled appointment service. For example if encology is scheduled, the click will keep hitting on the word encology which is `<div class="fc-event-inner">.`  whereas a plain day block without an appointment is this `<td class="fc-day fc-tue fc-widget-content fc-today fc-state-highlight" data-date="date-today">...</td>`. So that is the cause of all my troubles. 

I am wondering why when the testpatient is deleted, the  scheduled appointment doesn't  get deleted too at the end of the test. Could someone have an idea? 

I am looking into another way I could implement this so that it does not break the build again. 

Any piece of advice is welcome.  Thank you.
